File: golang/problem006.go
package main

import (
	"euler_math"
	"fmt"
)

func main() {
	count := 100
	soSquare := euler_math.SumOfSquares(count)
	soSums := euler_math.SquareOfSums(count)

	fmt.Println("Sum of squares: ", soSquare)
	fmt.Println("Square of sums: ", soSums)

	fmt.Println("Difference: ", soSums-soSquare)
}

func sumOfSquares(count int) int {
	sum := 0
	for i := 1; i <= count; i++ {
		sum += i * i
	}

	return sum
}

func squareOfSums(count int) int {
	sum := 0

	for i := 1; i <= count; i++ {
		sum += i
	}

	return sum * sum
}

File: C#/EulerQuestion/Problem006.cs
using System;

namespace EulerQuestion
{
    public class Problem006 : IProblem
    {
		public string Name { get { return "Sum square difference"; } }

        public IResult Question()
        {
            var begin = DateTime.Now.Ticks;
            return new Result
            {
				Name = this.Name,
                Answer = MathHelper.SquareOfSums(100) - MathHelper.SumOfSquares(100),
                GivenAnswer = 25164150,
                TimeTaken = new TimeSpan(DateTime.Now.Ticks - begin)
            };
        }

        public IResult Sample()
        {
            var begin = DateTime.Now.Ticks;
            return new Result
            {
				Name = this.Name,
                Answer = MathHelper.SquareOfSums(10) - MathHelper.SumOfSquares(10),
                GivenAnswer = 2640,
                TimeTaken = new TimeSpan(DateTime.Now.Ticks - begin)
            };
        }
    }
}
